nextstrain/cli/command/check_setup.py

```diff
@@ -35,16 +35,20 @@
 """
 
 from functools import partial
-from .. import config
 from ..argparse import SKIP_AUTO_DEFAULT_IN_HELP, runner_module_argument
 from ..types import Options
-from ..util import colored, check_for_new_version, runner_name, runner_tests_ok, print_runner_tests
+from ..util import colored, check_for_new_version, runner_name, setup_tests_ok, print_setup_tests
 from ..runner import all_runners, all_runners_by_name, default_runner # noqa: F401 (it's wrong; we use it in run())
 
 
 __doc__ = (__doc__ or "").format(default_runner_name = runner_name(default_runner))
 
 
+# XXX TODO: Add support for checking pathogen setups too?  Not sure this makes
+# much sense.
+#   -trs, 3 March 2025
```

Comment:
_non-blocking_

FWIW, I do think it'd be helpful to check existing pathogen set ups to remember which versions are fully functional. (But maybe that's only true right now when most of our pathogen repos have not been ported over?)

Reply:
Heh. I do think that will be less true later. OTOH, it will continue to be useful to re-run checks in the face of updated in-development versions. I guess that would be more reason to run checks after update

https://github.com/nextstrain/cli/blob/b382b59eade72c4157a1688d82a26b75cefe5c8c/nextstrain/cli/pathogens.py#L528-L532

Ah, a realization! The Docker/Singularity/AWS Batch runtime checks focus on the machinery required to use the runtime image, but do not test the contents/correctness of the runtime image itself; they assume that's ok. The pathogen checks OTOH solely test the contents/correctness of the pathogen itself; so it makes sense to check it after any change/update unlike the Docker checks. The Conda runtime checks do both!

Reply:
> FWIW, I do think it'd be helpful to check existing pathogen set ups to remember which versions are fully functional

Also, FWIW, to re-check existing setups you can also re-run `nextstrain setup` (without `--force`, of course).

Reply:
> I guess that would be more reason to run checks after update

+1 for checks after update, which then makes me think it'd be nice to have a way to "revert" an update if it's borked.

Reply:
Totally. Reverting should only be necessary, though, in the case that you've setup and updated a mutable version, e.g. a branch name or moving tag. If you're updating a default that's tracking a released version (i.e. static tags), an update ends up setting up the newest release version alongside the existing one and then changing the default. And adding checks after update, as we want to, would prevent the default from being set if they didn't pass.

Reply:
Revert will wait for later (maybe as part of [rolling back](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1743548815250379?thread_ts=1743533445.065599&cid=C01LCTT7JNN) failed setups?), but checks-on-update is now part of e2ed56a with the [latest repush](https://tsibley.net/paste/2025-04-02UtRuFy1x-diff-rebase.html).

![image](https://github.com/user-attachments/assets/94db6815-ab74-49ab-8a9c-15b85797879a)